src/ClassSchemaGenerator/ClassSchemaGenerator.php

```
@@ -382,4 +386,33 @@ private function getDiscriminatorMapSchema(string $propertyName, array $mapping,
+            return false;
+        }
+
+        // Property does not have a default value - check if we have the flag for making nullable properties not
+        // required and if we have a null as a possible type
+        if (
+            $this->configuration->considerNullablePropertiesAsNotRequired
+            && (
+                Schema\Type::NULL === $propertySchema->type
+                || (
+                    is_array($propertySchema->type)
+                    && array_any($propertySchema->type, static fn ($type) => Schema\Type::NULL === $type)
```

PHPStan (2.0.3)

Function array_any not found.
Learn more at https://phpstan.org/user-guide/discovering-symbols

(function.notFound)